var/spack/repos/builtin/packages/py-hypothesis/package.py

```diff
@@ -21,9 +21,11 @@ class PyHypothesis(PythonPackage):
     # TODO: Add missing dependency required to import hypothesis.extra.django
 
     version('4.41.2', sha256='6847df3ffb4aa52798621dd007e6b61dbcf2d76c30ba37dc2699720e2c734b7a')
+    version('4.24.3', sha256='fd90a319f409f34a173156ca704d6c0c6c0bb30a2e43dbf26aced2c75569e5d5')
     version('4.7.2',  sha256='87944c6379f77634474b88abbf1e5ed5fe966637cc926131eda5e2af5b54a608')
     version('3.7.0',  sha256='0fea49d08f2d5884f014151a5af6fb48d862f6ad567ffc4a2e84abf2f186c423')
 
     depends_on('py-setuptools', type='build')
-    depends_on('py-attrs@19.2.0:', type=('build', 'run'))
+    depends_on('py-attrs@19.2.0:', type=('build', 'run'), when='@4.41.2:')
+    depends_on('py-attrs@16.0.0:', type=('build', 'run'), when='@:4.41.1')
```

Comment:
I think this should be:
```python
depends_on('py-attrs@19.2.0:', when='@4.38.2:', type=('build', 'run'))
depends_on('py-attrs@16.0.0:', when='@3.44.22:', type=('build', 'run'))
depends_on('py-attrs', when='@3.28.0:', type=('build', 'run'))
```

See https://github.com/HypothesisWorks/hypothesis/releases/tag/hypothesis-python-4.38.2 and https://github.com/HypothesisWorks/hypothesis/commit/6e7851e97593e9d03c75cfe3690abf90c69f1f46#diff-2eeaed663bd0d25b7e608891384b7298 and https://github.com/HypothesisWorks/hypothesis/commit/c10b975ea9f1178592f1044eb1d69c3a4c4a482f#diff-2eeaed663bd0d25b7e608891384b7298